From 15a4cc1ab2f36f3d56458bd52ffd11cd3e2642df Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Wed, 24 Jul 2024 11:29:18 -0700
Subject: [PATCH] wifi: ath11k: advertise support for GreenAP

Advertise driver support for GreenAP to mac80211 for IPQ9574.
Replace BSS_CHANGED_PS by BSS_CHANGED_AP_PS which is used for AP mode.

Patch-dependency: 909-Add-the-new-flag-for-Green-AP-ps-mode.patch

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/core.c | 1 +
 drivers/net/wireless/ath/ath11k/mac.c  | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath11k/core.c
+++ b/drivers/net/wireless/ath/ath11k/core.c
@@ -989,6 +989,7 @@ static struct ath11k_hw_params ath11k_hw
 		.max_tx_ring = DP_TCL_NUM_RING_MAX + 1,
 		//.wakeup_mhi = false,
 		.num_vdevs_peers = ath11k_vdevs_peers,
+		.supports_ap_ps = true,
 		.supports_sta_ps = false,
 		.dbr_debug_support = true,
 		.bios_sar_capa = NULL,
--- a/drivers/net/wireless/ath/ath11k/mac.c
+++ b/drivers/net/wireless/ath/ath11k/mac.c
@@ -4064,7 +4064,7 @@ static void ath11k_mac_op_bss_info_chang
 
 	ath11k_mac_fils_discovery(arvif, info);
 
-	if ((changed & BSS_CHANGED_PS) && vif->type == NL80211_IFTYPE_AP) {
+	if ((changed & BSS_CHANGED_AP_PS) && vif->type == NL80211_IFTYPE_AP) {
 		ar->ap_ps_enabled = info->ap_ps_enable;
 		ath11k_mac_ap_ps_recalc(ar);
 	}
